Compile error on Swift 4.2

With Swift 4.2, I get the following error:

```
/Users/pcmoritz/XXX/.build/checkouts/url-encoded-form.git--6514778735335937147/Sources/URLEncodedForm/Data/URLEncodedFormParser.swift:101:44: error: ambiguous use of 'init'
            subKeys = try slices[1...].map(Data.init).map { data -> URLEncodedFormEncodedSubKey in
                                           ^
Foundation.Data:29:23: note: found this candidate
    @inlinable public init<S>(_ elements: S) where S : Sequence, S.Element == UInt8
                      ^
Foundation.Data:30:23: note: found this candidate
    @inlinable public init<S>(bytes elements: S) where S : Sequence, S.Element == UInt8
                      ^
error: terminated(1): /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2018-07-12-a.xctoolchain/usr/bin/swift-build-tool -f /Users/pcmoritz/XXX/.build/debug.yaml main output:
```

--- a/Sources/URLEncodedForm/Data/URLEncodedFormParser.swift
+++ b/Sources/URLEncodedForm/Data/URLEncodedFormParser.swift
@@ -98,12 +98,14 @@ final class URLEncodedFormParser {
                 throw URLEncodedFormError(identifier: "malformedKey", reason: "Malformed form-urlencoded key encountered.")
             }
             stringData = Data(slices[0])
-            subKeys = try slices[1...].map(Data.init).map { data -> URLEncodedFormEncodedSubKey in
-                if data[0] == .rightSquareBracket {
-                    return .array
-                } else {
-                    return try .dictionary(data.dropLast().utf8DecodedString())
-                }
+            subKeys = try slices[1...]
+                .map { Data($0) }
+                .map { data -> URLEncodedFormEncodedSubKey in
+                    if data[0] == .rightSquareBracket {
+                        return .array
+                    } else {
+                        return try .dictionary(data.dropLast().utf8DecodedString())
+                    }
             }
         } else {
             stringData = data

